pkg/api/types.go

```diff
@@ -252,6 +255,9 @@ type PersistentVolumeSpec struct {
 	AccessModes []AccessModeType `json:"accessModes,omitempty"`
 	// holds the binding reference to a PersistentVolumeClaim
 	ClaimRef *ObjectReference `json:"claimRef,omitempty"`
+	// SecretRef is a reference to a Secret in the system.
+	// used for persistent volume authentication
+	SecretRef *ObjectReference `json:"secretRef,omitempty"`
```

Comment:
At what point is the secretRef expected to be assigned a value? When the PersistentVolume is created? When the claim is bound?